### DISCUSSION TITLE
problem to connect in s3

### DISCUSSION THREAD

**Discussion by lokize** [pocketbase/pocketbase#801] [Q&A] [answered]
hi, i have a problem to connect in s3:

![image](https://user-images.githubusercontent.com/19701630/196520495-9ca30ce9-3a48-473c-a652-1821bce92891.png)

**Comment by ganigeorgiev** [comment, marked as answer]
Are you running inside a docker? If you are using a stripped-down base image you may have to install `ca-certificates`.

**Reply by lokize** [reply]
thanks for reply

yes running in docker, how to install ca-certificates?

**Reply by ganigeorgiev** [reply]
This depends on the package manager that your base docker image is using. `Alpine` for example uses `apk`. Here is an example Dockerfile that install `ca-certificates`:

```dockerfile
FROM alpine:latest

ARG PB_VERSION=0.7.9

RUN apk add --no-cache \
    unzip \
    ca-certificates

# download and unzip PocketBase
ADD https://github.com/pocketbase/pocketbase/releases/download/v${PB_VERSION}/pocketbase_${PB_VERSION}_linux_amd64.zip /tmp/pb.zip
RUN unzip /tmp/pb.zip -d /pb/

EXPOSE 8080

# start PocketBase
CMD ["/pb/pocketbase", "serve", "--http=0.0.0.0:8080"]
```

**Reply by lokize** [reply]
its worked! thank you very much!

**Reply by im-ramon** [reply]
> This depends on the package manager that your base docker image is using. `Alpine` for example uses `apk`. Here is an example Dockerfile that install `ca-certificates`:
> 
> ```dockerfile
> FROM alpine:latest
> 
> ARG PB_VERSION=0.7.9
> 
> RUN apk add --no-cache \
>     unzip \
>     ca-certificates
> 
> # download and unzip PocketBase
> ADD https://github.com/pocketbase/pocketbase/releases/download/v${PB_VERSION}/pocketbase_${PB_VERSION}_linux_amd64.zip /tmp/pb.zip
> RUN unzip /tmp/pb.zip -d /pb/
> 
> EXPOSE 8080
> 
> # start PocketBase
> CMD ["/pb/pocketbase", "serve", "--http=0.0.0.0:8080"]
> ```
I'm trying too long time, even following these guidelines, I can't send emails on the local machine. Am I still doing something wrong?

Dockerfile:
```Dockerfile
FROM alpine:latest

ARG PB_VERSION=0.11.3

RUN apk add --no-cache \
    unzip \
    ca-certificates

RUN apk update
RUN addgroup -S pocketbase && adduser -S pocketbase -G pocketbase

ADD https://github.com/pocketbase/pocketbase/releases/download/v${PB_VERSION}/pocketbase_${PB_VERSION}_linux_amd64.zip /tmp/pb.zip
RUN unzip /tmp/pb.zip -d /pb/
RUN mkdir pb/pb_data

RUN chown -R pocketbase:pocketbase /pb
RUN chmod -R 710 /pb

USER pocketbase

VOLUME /pb/pb_data
EXPOSE 8090
CMD ["/pb/pocketbase", "serve", "--http=0.0.0.0:8090"]
```

docker-compose.yml:
```docker-compose
version: '3.7'

services:
  pocketbase:
    build: .
    restart: always
    ports:
      - 8090:8090
    volumes:
      - ./pb_data:/pb/pb_data
```

**Reply by ganigeorgiev** [reply]
@im-ramon What is the error you are getting when sending test emails?
Note that by default PocketBase uses the `sendmail` command (if available) for sending emails which is not reliable and it is recommended to use a SMTP server (for test purposes I often use [mailtrap.io](https://mailtrap.io/)).
If that's not the case, please provide more details and/or open a new discussion.
